### PR TITLE
created endpoint to send user input (job description/resume) to the api

### PR DESCRIPTION
**Pull request by haitham-awad**

### CODE REVIEW

**Comment by haitham-awad** [comment]
[Task 18: Develop an API Endpoint to Send Resume and Job Description to the NLP API](https://trello.com/c/7SbAWr7n/22-task-18-develop-an-api-endpoint-to-send-resume-and-job-description-to-the-nlp-api)

File: backend/utils/natural_language_processing_api.py [hunk 1]
```diff
@@ -41,4 +41,10 @@ def prompt_nlp_model(prompt):
...
 
     # return the information into a json format
-    return json.loads(api_prompt.model_dump_json(indent=2))
+    text = api_prompt.choices[0].message.content
+    json_response = json.loads(text)
+
+    if not json_response["fit_score"] or not json_response["feedback"]:
+        return {"error" : "model response not in desired format"}
+
+    return json.loads(text)
```
Comment (arikamat) [inline backend/utils/natural_language_processing_api.py]: can we consider importing the dictionary as a pydantic model to enforce the way our output is structured.
```python
class FitScore(BaseModel):
    fit_score: int
    feedback: List[str]
```

```python
res = FitScore(**json_response)
return res
```

Comment (arikamat) [inline backend/utils/natural_language_processing_api.py]: `json.loads(text)` being run twice

**Review by arikamat — CHANGES_REQUESTED** [review]

**Review by npaul64 — DISMISSED** [review]

File: backend/utils/natural_language_processing_api.py [hunk 2]
```diff
@@ -40,5 +46,8 @@ def prompt_nlp_model(prompt):
     except APIConnectionError:
         return {"error": "No API KEY provided"}
 
-    # return the information into a json format
-    return json.loads(api_prompt.model_dump_json(indent=2))
+    # if organizing the response into FitScore doesn't work
+    except json.decoder.JSONDecodeError:
+        return {"error": "API Response was not in the correct format"}
```
Comment (arikamat) [inline backend/utils/natural_language_processing_api.py:52]: you should also catch 
```python
ValidationError
```
This is when the json_response can't be fit into FitScore model (due to improper format)

File: tests/backend/test_utils_natural_language_processing_api.py
Comment (arikamat) [inline tests/backend/test_utils_natural_language_processing_api.py:1]: Check the code coverage for this file. There are a bunch of tests that you seem to be missing

**Review by arikamat — CHANGES_REQUESTED** [review]

**Review by arikamat — APPROVED** [review]

**Comment by haitham-awad** [comment]
[Task 19: Create Standardized Input/Output Data Structures to Communicate with the AI API](https://trello.com/c/Syxg9GHx/23-task-19-create-standardized-input-output-data-structures-to-communicate-with-the-ai-api)

**Review by npaul64 — APPROVED** [review]

**Pull request merged**